exercises/collatz-conjecture/test/Tests.hs

```diff
@@ -0,0 +1,59 @@
+{-# OPTIONS_GHC -fno-warn-type-defaults #-}
+{-# LANGUAGE RecordWildCards #-}
+
+import Data.Foldable     (for_)
+import Test.Hspec        (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+
+import CollatzConjecture (collatz)
+
+main :: IO ()
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "collatz" $ for_ cases test
+  where
+
+    test Case{..} = it description assertion
+      where
+        assertion = collatz number `shouldBe` expected
+
+
+data Case = Case { description :: String
+                 , property    :: String
+                 , number      :: Integer
+                 , expected    :: Maybe Integer
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "zero steps for one"
+               , property    = "steps"
```

Comment:
the `property` is not being used. It should be removed.

Reply:
i will, but maybe it has to be removed from [https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json](https://github.com/exercism/problem-specifications/blob/master/exercises/collatz-conjecture/canonical-data.json)

Reply:
It needs to be in problem-specifications since it tells what is being tested.

It must not be here since we already know what we are testing (it's specified in code)